src/features/More/components/categoryItem.tsx

```diff
@@ -11,52 +13,111 @@ interface CategoryItemProps {
 }
 
 const CategoryItem = (props: CategoryItemProps) => {
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  const handlePress = () => {
+    props.setIndexOfPressedCategory(props.itemData.index);
+    setIsExpanded(!isExpanded);
+  };
+
+  useEffect(() => {
+    if (!props.isSelected) {
+      setIsExpanded(false);
+    }
+  }, [isExpanded, props.isSelected]);
+
   return (
-    <TouchableOpacity
-      style={{
-        ...styles.listItem,
-        backgroundColor: props.isSelected ? Colors.ACTIVE_ICON : Colors.WHITE,
-      }}
-      onPress={() => {
-        props.setIndexOfPressedCategory(props.itemData.index);
-      }}
-    >
-      <View style={styles.itemContainer}>
-        <View style={styles.arrowIcon}>
-          <ArrowIcon />
-        </View>
-        <View>
-          <Text style={styles.title}>{props.itemData.item?.title}</Text>
+    <View>
+      <TouchableOpacity
+        style={{
+          ...styles(isExpanded).listItem,
+          backgroundColor: props.isSelected ? Colors.ACTIVE_ICON : Colors.WHITE,
+        }}
+        onPress={handlePress}
+      >
+        <View style={styles(isExpanded).itemContainer}>
+          {props.itemData.item.children.length !== 0 && (
+            <View style={styles(isExpanded).arrowIcon}>
+              <ArrowIcon
+                transform={
+                  isExpanded
+                    ? Strings.moreScreen.UP_ARROW
+                    : Strings.moreScreen.DOWN_ARROW
+                }
+              />
+            </View>
+          )}
+
+          <View style={styles(isExpanded).titleContainer}>
+            <Text style={styles(isExpanded).title}>
+              {props.itemData.item?.title}
+            </Text>
+          </View>
         </View>
+      </TouchableOpacity>
+
+      <View style={styles(isExpanded).subCategories}>
```

Comment:
Its better but what I suggested is doing:
`props.itemData.item.children.length > 0 && ( <View style={styles(isExpanded).subCategories}>....</View>`

This is conditional rendering... if the first condition is not true it will not do the second one and therefore nothing will be render...